iroh-store/src/config.rs

```diff
@@ -17,6 +17,15 @@ pub const CONFIG_FILE_NAME: &str = "store.config.toml";
 /// For example, `IROH_STORE_PATH=/path/to/config` would set the value of the `Config.path` field
 pub const ENV_PREFIX: &str = "IROH_STORE";
 
+/// the path to data directory. If arg_path is `None`, the default iroh_data_path()/store is used
+/// iroh_data_path() returns an operating system-specific directory
+pub fn config_data_path(arg_path: Option<PathBuf>) -> Result<PathBuf> {
+    match arg_path {
+        Some(p) => Ok(p),
+        None => iroh_data_path("store"),
+    }
```

Comment:
I'm unclear on weather or not there's a more succinct way to express this

Reply:
I think `arg_path.map_or(iroh_data_path("store"), |v| Ok(v))` is equivalent - but not easier to read!

Reply:
oooh I hadn't found `map_or`. Thanks for the tip @fabricedesre! I agree with you that the  `match` is easier to read, and will leave it as is unless I'm overruled by dig 😄

Reply:
seems fine and readable